pkg/testing/integration/program.go

```diff
@@ -1107,6 +1166,49 @@ func (pt *programTester) prepareNodeJSProject(projinfo *engine.Projinfo) error {
 
 // preparePythonProject runs setup necessary to get a Python project ready for `pulumi` commands.
 func (pt *programTester) preparePythonProject(projinfo *engine.Projinfo) error {
+	cwd, _, err := projinfo.GetPwdMain()
+	if err != nil {
+		return err
+	}
+
+	// Create a new Pipenv environment. This bootstraps a new virtual environment containing the version of Python that
+	// we requested. Note that this version of Python is sourced from the machine, so you must first install the version
```

Comment:
We allow the version of Python to be configured per-test.  Should we?  Is there a well known way to install multiple Python 3.X versions concurrently?

Reply:
> We allow the version of Python to be configured per-test. Should we? Is there a well known way to install multiple Python 3.X versions concurrently?

Yeah, it's possible and not uncommon to do this. There's even a [PEP](https://www.python.org/dev/peps/pep-0394/) to document a convention for having different versions of Python on the path. In general there's nothing stopping people from having multiple `python3.x` executables on their path pointing to different Python versions. It's certainly possible to do so; the Python toolchain has evolved around the necessity to install Python 2 and 3 side-by-side so it's actually not to hard to do this today with Python 3 versions.

As to whether or not we should let tests select their version of Python, I doubt it's going to be useful in the near term. I don't mind removing it.